### PR TITLE
[LISAv3] Adding container test suite

### PR DESCRIPTION
**Pull request by isaiahduncan**
Adding Container Test Suite. 
New tool being created for the test suite.

### CODE REVIEW

File: lisa/tools/__init__.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from .chrony import Chrony
 from .date import Date
 from .dmesg import Dmesg
+from .docker import Docker
```
Comment (squirrelsc) [inline lisa/tools/__init__.py]: It's better to put the tool change into a separated commit.

File: microsoft/testsuites/container/dockersuite.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            2. x
+        """,
+        priority=2,
+    )
+    def _evaluate_test_results():
+        docker_tool = node.tools[Docker]
+        pass
+
+
+    def before_suite(self, **kwargs: Any) -> None:
```
Comment (squirrelsc) [inline microsoft/testsuites/container/dockersuite.py]: If there is no content in those method, they don't need to be added.

File: microsoft/testsuites/container/dockersuite.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        description="""
+            This test case ...
+
+            Steps:
+            1. x
+            2. x
+        """,
+        priority=2,
+    )
+    def _generate_java_program():
```
Comment (squirrelsc) [inline microsoft/testsuites/container/dockersuite.py]: If they are mock up methods. You can submit them next time for reviewing.

File: lisa/tools/docker.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa.executable import Tool
+from lisa.operating_system import CentOs,Debian, Redhat, Ubuntu
+
+class Docker(Tool):
+
+    def _install_docker_engine(self):
```
Comment (squirrelsc) [inline lisa/tools/docker.py]: The tool support a _install method. Refer to document and other tools about how they implemented.

File: lisa/tools/docker.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        [[ $ret -ne 0 ]] && return $ret
+
+        VerifyDockerEngine; ret=$?
+        [[ $ret -ne 0 ]] && return $ret
+
+        return $ret
+    }
+    """
+
+    def _build_docker_image():
```
Comment (squirrelsc) [inline lisa/tools/docker.py]: It looks the code is not ready to review. You can work on your branch, and submit PR when it's ready to merge.

**Comment by isaiahduncan** [comment]
@mcgov could you please take a look at the _install_docker_engine function in docker.py?

**Pull request closed without merging**